cmd/agentctl/main.go

```diff
@@ -84,6 +86,36 @@ source-of-truth directory.`,
 	return cmd
 }
 
+func configCheckCmd() *cobra.Command {
+	var expandEnv bool
+
+	cmd := &cobra.Command{
+		Use:   "config-check [config file]",
+		Short: "Perform basic validation of the given Agent configuration file",
+		Long: `config-check performs basic syntactic validation of the given Agent configuration
+file. The file is checked to ensure the types match the expected configuration types. Optionally,
+${var} style substitutions can be expanded based on the values of the environmental variables.
+
+If the configuration file is valid the exit code will be 0. If the configuration file is invalid
+the exit code will be 1.`,
+		Args: cobra.ExactArgs(1),
+		Run: func(_ *cobra.Command, args []string) {
+			file := args[0]
+			logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
+
+			cfg := config.Config{}
+			err := config.LoadFile(file, expandEnv, &cfg)
+			if err != nil {
+				level.Error(logger).Log("msg", "failed to validate config", "err", err)
+				os.Exit(1)
+			}
+		},
```

Comment:
WDYT about printing out `config valid` here? The exit code will be the most useful for validation, but I think it will be nicer from the user's perspective if it doesn't silently succeed

Reply:
WFM. Do you think it makes sense to `fmt.Fprintf` or log at info?

Reply:
`Fprintf` to stdout for the success makes the most sense to me in this case